var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -321,14 +320,17 @@ class Openmpi(AutotoolsPackage, CudaPackage):
     depends_on('lsf', when='schedulers=lsf')
     depends_on('pbs', when='schedulers=tm')
     depends_on('slurm', when='schedulers=slurm')
-    depends_on('libevent', when='+pmix')
 
-    # PMIx is unavailable for @1, an option for @2:4 and required for @5:
-    # In @4, if it's not disabled (and PMI is not explicitly enabled), an
-    # internal copy will be used instead
-    # Vendored version: depends_on('pmix@3.2.3', when='@4.1.2')
-    depends_on('pmix', when='+pmix')
-    depends_on('pmix@3.2:', when='@4.0:4 +pmix')
+    # PMIx is unavailable for @1, and required for @2:
+    # OpenMPI @2: includes a vendored version:
+    # depends_on('pmix@1.1.2', when='@2.1.6')
+    # depends_on('pmix@3.2.3', when='@4.1.2')
+    depends_on('pmix@1.0:1', when='@2.0:2')
+    depends_on('pmix@3.2:', when='@4.0:4')
+    depends_on('pmix@5:', when='@5.0:5')
+
+    # Libevent is required for PMIx
+    depends_on('libevent', when='@2:')
```

Comment:
```suggestion
    depends_on('libevent', when='@main')
```

Reply:
I merged this with a rogue `libevent@2:` so the logic should be correct now